rasa/core/evaluation/marker.py

```diff
@@ -131,12 +149,27 @@ def negated_tag() -> Optional[Text]:
         """Returns the tag to be used in a config file for the negated version."""
         return "action_not_executed"
 
+    def validate_against_domain(self, domain: Domain) -> bool:
+        """Checks that this marker (and its children) refer to entries in the domain.
+
+        Args:
+            domain: The domain to check against
+        """
+        valid = self.text in domain.action_names_or_texts
+
+        if not valid:
+            rasa.shared.utils.io.raise_warning(
+                f"Referenced action '{self.text}' does not exist in the domain"
+            )
```

Comment:
@usc-m and @ka-bu I was creating the QA tests and was curious why this (and the same for intent and slot) is a warning and not an error. Does this mean that the extraction process will continue if there is a mismatch with the domain? What's the advantage of allowing this?

Reply:
The extraction process isn't managed here - this goes over each marker and decides if they are valid, and if not emits a message, and then passes back True/False if it's valid or not. In the CLI we catch that boolean and error and exit based on it, but in these I wanted a way to collect _all_ of the issues without stopping first, and then stop later (since reporting one at a time would be frustrating for users). The reason it's a warning is mostly because that was my first pass on it. Ideally it would go out as an error without terminating (and then get caught in the CLI to wrap it up and actually terminate) but we don't have a helper function for that so it's a little more involved - definitely something I want to fix before this merges.

Reply:
The CLI also terminates before the extraction process begins. It loads the marker definitions and the domain, does the validation first, and terminates if issues are found. It will only establish a connection to the tracker store (and by extension perform extraction) if there are no issues

Reply:
or in short: https://github.com/RasaHQ/rasa/pull/10012#discussion_r740836126 :D

Reply:
Looked at how DialogueStateTracker manages this and have replicated

Reply:
I like that it's consistent but that approach is also a bit confusing, isn't it? I would've expected a "raise..." if there's a logger error (and usually, I would not explicitly log the error but just have a decorator to pass on the error message)

Reply:
... on the other hand, getting an error as a result of "only warnings" is also a bit unexpected maybe 😅

Reply:
Yeah, I figured it was consistent from the perspective of the user, but it might be worth re-evaluating how we do error management across the codebase at some point